### DISCUSSION TITLE
Filter current date between startDate and endDate

### DISCUSSION THREAD

**Discussion by MartinPELCAT** [prisma/prisma#12767] [Q&A] [answered]
Hello, is there any way to make a query to get an entry that have the current date between startDate and endDate ?

What i want :
- The next appointement if currentDate is not between any startDate and endDate
- Or the current appointement if currentDate is between the startDate and endDate 

For now that's what i have
Do i have to use a "queryRaw" ?

```typescript
const currentDate = new Date()
return db.appointment.findFirst({
  where: {
    AND: [
      {
        patientUuid: user.uuid,
      },
      {
        date: {
          gte: currentDate,
        },
      },
    ],
  },
  orderBy: {
    date: "asc",
  },
  rejectOnNotFound: false,
});

```

```prisma
model Appointment {
id @id Int
startDate DateTime @db.Timestamptz
endDate DateTime @db.Timestamptz
}
```

**Comment by MartinPELCAT** [comment]
Could this work ??

```typescript
const currentDate = new Date();
return db.appointment.findFirst({
  where: {
    AND: [
      {
        patientUuid: user.uuid,
      },
      {
        OR: [
          {
            AND: [
              {
                date: {
                  lte: currentDate,
                },
              },
              {
                endDate: {
                  gte: currentDate,
                },
              },
            ],
          },
          {
            date: {
              gte: currentDate,
            },
          },
        ],
      },
    ],
  },
  orderBy: {
    date: "asc",
  },
  rejectOnNotFound: false,
});

```

**Comment by pantharshit00** [comment, marked as answer]
Hello @MartinPELCAT , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

The second query you have posted should work here. Let me know if it didn't produce the desired result.

**Reply by vladi-stevanovic** [reply]
👋 Hello @MartinPELCAT I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if your last query already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by MartinPELCAT** [reply]
Thank you, yes this issue can be closed i figured out a way to achieve what i wanted!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
